onnx_xla/operator_registry.cc

```diff
@@ -0,0 +1,51 @@
+#include "onnx_xla/operator_registry.h"
+#include <iostream>
+namespace onnx_xla  {
+  
+  OperatorRegistry::OperatorRegisterOnce::OperatorRegisterOnce(const Symbol& nodeKind, TranslationFunction translator)  {
+    auto& map = OperatorRegistry::map();
+    if (map.find(nodeKind) == map.end())  {
+      map[nodeKind] = translator;
+    } else {
+      throw std::runtime_error("Operator registry error");
+    }
+  }
+
+  onnxStatus OperatorRegistry::executeTranslation(Node& n, XlaBuilder& builder, ValueOpMap& valueToOp)  {
+    auto& map = OperatorRegistry::map();
+    if (map.find(n.kind()) != map.end())  {
+      return map[n.kind()](n, builder, valueToOp);
+    } else {
+      return ONNXIFI_STATUS_UNSUPPORTED_OPERATOR;
+    }
+  }
+
+  OperatorRegistry* OperatorRegistry::registry()  {
+    static OperatorRegistry registry_;
+    return &registry_;
+  }
+
+  TranslationMap& OperatorRegistry::map()  {
+    static TranslationMap map;
+    return map;
+  }
+
+  onnxStatus translateRelu(Node& n, XlaBuilder& builder, ValueOpMap& valueToOp)  {
+    auto input = valueToOp[n.inputs()[0]];
```

Comment:
should be using `at` (instead of `[]`) here?

Reply:
Indeed safer with at with bounds checking, so changed - but i suppose the onnx model checker would catch an error if the relu node does not have right # of inputs?